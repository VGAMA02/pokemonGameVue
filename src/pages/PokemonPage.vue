<template>

    <h1 v-if="!pokemon"> </h1>
    <div v-else>
        <h1>Â¿Quien es este pokemon?</h1>
        <!--IMG -->
        <PokemonPictureVue :pokemonId="pokemon.id" :showPokemon="showPokemon" />
        <!-- Opciones -->
        <PokemonOptionsVue :pokemons="pokemonArr" @selection="checkAnswer"/>

        <div v-if="showAnswer">
            <h2>{{message}}</h2>
            <button @click="newGame">Nuevo juego</button>
        </div>


    </div>

</template>

<script>
import PokemonOptionsVue from '@/components/PokemonOptions.vue'
import PokemonPictureVue from '@/components/PokemonPicture.vue'
import getPokemonOptions from '@/helpers/getPokemonOptions'
/* console.log(getPokemonOptions()); */
export default {
  components: {
    PokemonOptionsVue,
    PokemonPictureVue
  },
  data(){
    return{
        pokemonArr:[],
        pokemon: null,
        showPokemon: false,
        showAnswer: false,
        message: ''
    }
  },
  methods:{
    async mixPokemonArray(){
        this.pokemonArr = await getPokemonOptions()
        const rndInt = Math.floor(Math.random() * 4)
        this.pokemon = this.pokemonArr[rndInt]
    },
    checkAnswer(selectId){
        this.showPokemon = true
        this.showAnswer = true
        if(selectId === this.pokemon.id){
            this.message = `Correcto, ${this.pokemon.name}`
        }else{
            this.message = `ups el pokemon es, ${this.pokemon.name}`
        }
    },
    newGame(){
        this.showPokemon = false
        this.showAnswer = false
        this.pokemonArr = []
        this.mixPokemonArray()
    }
  },
  mounted(){
    this.mixPokemonArray()
  }
}
</script>

<style>

</style>