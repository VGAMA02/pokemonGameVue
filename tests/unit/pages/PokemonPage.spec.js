
import { shallowMount } from '@vue/test-utils'
import PokemonPage from '@/pages/PokemonPage'
import { pokemons } from '../mocks/pokemons.mock'
describe('Pokemon Component', () => {
    let wrapper

    beforeEach(() => {
        wrapper = shallowMount(PokemonPage)
        
    })
    test('debe de hacer match con el snapsshot',() => {
        expect(wrapper.html()).toMatchSnapshot()
    })
    test('debe de llamar mixPokemonArray ak montar',() => {
        const mixPokemonArraySpy = jest.spyOn(PokemonPage.methods, 'mixPokemonArray')
        const wrapper = shallowMount(PokemonPage) // se monta de nuevo para que el spy funcione
        expect(mixPokemonArraySpy).toHaveBeenCalled()
    })
    test('debe de hacer match con el snapsshot cuando cargan los pokemons',() => {
        const wrapper = shallowMount(PokemonPage, {
            data(){
                return{
                    pokemonArr:pokemons,
                    pokemon: pokemons[0],
                    showPokemon: false,
                    showAnswer: false,
                    message: ''
                }
            }
        })
        expect(wrapper.html()).toMatchSnapshot()
    })

    test('debe de mostrar los componentes pokemonPicture y pokemonOptions',() => {
        const wrapper = shallowMount(PokemonPage, {
            data(){
                return{
                    pokemonArr:pokemons,
                    pokemon: pokemons[0],
                    showPokemon: false,
                    showAnswer: false,
                    message: ''
                }
            }
        })
        
        expect(wrapper.find('pokemon-picture-vue-stub').exists()).toBeTruthy()
        expect(wrapper.find('pokemon-options-vue-stub').exists()).toBeTruthy()
        expect(wrapper.find('pokemon-picture-vue-stub').attributes('pokemonid')).toBe("1")
        expect(wrapper.find('pokemon-options-vue-stub').attributes('pokemons')).toBeTruthy()
    }) 

    test('pruebas con check answer', async() => {
        const wrapper = shallowMount(PokemonPage, {
            data(){
                return{
                    pokemonArr:pokemons,
                    pokemon: pokemons[0],
                    showPokemon: false,
                    showAnswer: false,
                    message: ''
                }
            }
        })

        await wrapper.vm.checkAnswer(1)
        expect(wrapper.find('h2').exists()).toBeTruthy()
        expect(wrapper.vm.showPokemon).toBeTruthy()
        expect(wrapper.vm.showPokemon).toBeTruthy()
        expect(wrapper.find('h2').text()).toBe(`Correcto, ${pokemons[0].name}`)
    }) 
})