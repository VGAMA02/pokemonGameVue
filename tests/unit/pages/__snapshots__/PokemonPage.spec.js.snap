// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pokemon Component debe de hacer match con el snapsshot 1`] = `<h1></h1>`;

exports[`Pokemon Component debe de hacer match con el snapsshot cuando cargan los pokemons 1`] = `
<div>
  <h1>Â¿Quien es este pokemon?</h1>
  <!--IMG -->
  <pokemon-picture-vue-stub pokemonid="1" showpokemon="false"></pokemon-picture-vue-stub><!-- Opciones -->
  <pokemon-options-vue-stub pokemons="[object Object],[object Object],[object Object],[object Object]"></pokemon-options-vue-stub>
  <!--v-if-->
</div>
`;
